xonsh/history.py

```diff
@@ -110,7 +110,8 @@ def files(self, only_unlocked=False):
         """
         _ = self  # this could be a function but is intimate to this class
         # pylint: disable=no-member
-        xdd = os.path.abspath(builtins.__xonsh_env__.get('XONSH_DATA_DIR'))
+        xdd = built_ins.expand_path(
```

Comment:
Sure, that make sense.  Sorry about the red herring.  Please revert this PR back to that state, and I'll go ahead and we can review again and merge.